cmd/iterate.go

```diff
@@ -111,6 +114,48 @@ func iterateStream(ctx context.Context, args []string, r io.Reader, fn func(ctx2
 	return nil
 }
 
+// varyBatch dynamically reduces the batch on document-exceeded-limit error and retries.
+func varyBatch(ctx context.Context, args []string, docs []json.RawMessage,
+	fn func(ctx2 context.Context, args []string, docs []json.RawMessage) error,
+) error {
+	first := 0
+	last := len(docs)
+
+	total := 0
+
+	for first < len(docs) {
+		if err := fn(ctx, args, docs[first:last]); err != nil {
+			if err.Error() == "document exceeds limit" && last-first > 1 {
```

Comment:
there is one for transaction size as well

Reply:
fixed.